server/src/main/java/org/opensearch/action/admin/indices/forcemerge/ForceMergeRequest.java

```diff
@@ -100,6 +102,9 @@ public ForceMergeRequest(StreamInput in) throws IOException {
         maxNumSegments = in.readInt();
         onlyExpungeDeletes = in.readBoolean();
         flush = in.readBoolean();
+        if (in.getVersion().onOrAfter(Version.V_2_12_0)) {
```

Comment:
also, don't we need to keep it 3.0.0 for main, i have seen some bwc test failures in past with keeping it as 2.x version. because i think tests uses the last minor version of previous version which will be 2.12.0 only.

Reply:
If this PR targets 2.12.0, I think we can keep it as `Version.V_2_12_0` because I don't see any bwc test failures except the rest yml test(I have set the supported version to 3.0.0), if we set it to 3.0.0, then after this PR has been merged and backported to 2.x branch, we need to change it back to 2.12.0 again because this parameter is introduced in 2.12.0. Inspired from https://github.com/opensearch-project/OpenSearch/pull/5503.